### PR TITLE
bpo-47167: Allow overriding a future compliance check in asyncio.Task

### PR DESCRIPTION
**Pull request by asvetlov**
https://bugs.python.org/issue47167

### CODE REVIEW

File: Modules/_asynciomodule.c [hunk 1]
```diff
@@ -2269,14 +2272,32 @@ Returns the remaining number of cancellation requests.
...
+static PyObject *
+_asyncio_Task__check_future_impl(TaskObj *self, PyObject *future)
+/*[clinic end generated code: output=2473f5b55d524bd5 input=76b4fa671a14b681]*/
+{
+    int res = task_check_future_exact(self, future);
+    if (res < 0) {
+        return NULL;
+    }
+    return PyBool_FromLong(res);
+}
```
Comment (arhadthedev) [inline Modules/_asynciomodule.c:2299]: It looks like the function can be simplified by using a `bool` return converter. It cannot be expressed in a simple diff so I [created a branch from your one](https://github.com/python/cpython/compare/asyncio-task-check...arhadthedev:_check_future-bool?expand=1).

Comment (arhadthedev) [inline Modules/_asynciomodule.c:2295]: It looks like the function can be simplified by using a `bool` return converter. It cannot be expressed in a simple diff so I [created a branch from your one](https://github.com/python/cpython/compare/asyncio-task-check...arhadthedev:_check_future-bool?expand=1).

Comment (asvetlov) [reply Modules/_asynciomodule.c:2295]: Thanks! The note is applyed

**Comment by 1st1** [comment]
I think this PR should be reverted. I explained why in the issue.

**Comment by 1st1** [comment]
@asvetlov Andrew, please don't force commits like this without my review (I'm still listed as a co-owner).

**Pull request merged**